src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -802,38 +794,6 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                         }
                     }
                     handleOne(callbacks, idx + 1);
-                } else if (callback instanceof SSLSessionAuthorizationCallback) {
-                    final SSLSessionAuthorizationCallback authorizationCallback = (SSLSessionAuthorizationCallback) callback;
-                    final SSLSession sslSession = authorizationCallback.getSslSession();
-                    final IdentityCache cache = (IdentityCache) SSLUtils.computeIfAbsent(sslSession, "org.wildfly.elytron.identity-cache", key -> new IdentityCache());
-                    final SecurityDomain securityDomain = stateRef.get().getSecurityDomain();
-                    final SecurityIdentity identity = cache.identities.get(securityDomain);
```

Comment:
@darranl I've added a commit that allow mechanisms to create an ```IdentityCache``` based on the current ```SecurityDomain```. 

Actually, that was a very interesting addition to the design as now mechanism have even more control on how their identities should be cached.